src/oba_from_uniswap/main.py

```diff
@@ -0,0 +1,227 @@
+# Postprocesses csv files obtained from dune analytics with routed trades
+
+import csv
+import argparse
+import ast
+import json
+from tqdm import tqdm
+from functools import lru_cache
+from frozendict import frozendict
+from fractions import Fraction as F
+from diskcache import Cache
+
+from ..subgraph import GraphQLClient, UniswapClient
+
+uniswap = UniswapClient()
+disk_cache = Cache(directory='.cache')
+
+def load_swaps(filename):
+    """Parse a csv file to a list of dicts."""
+    r = []
+    with open(filename, "r") as f:
+        reader = csv.reader(f)
+        first = True
+        for row in reader:
+            # Skip header.
+            if first:
+                first = False
+                continue
+            (block_number, index, sell_amount,
+            buy_amount, path, output_amounts) = row
+
+            path = ast.literal_eval(path)
+            path = ['0x' + address for address in path]
+            output_amounts = ast.literal_eval(output_amounts.replace('L',''))
+            output_amounts = list(map(int, output_amounts))
+            r.append({
+                'block_number': int(block_number),
+                'index': int(index),
+                'sell_amount': int(sell_amount),
+                'buy_amount': int(buy_amount),
+                'path': path,
+                'output_amounts': output_amounts
+            })
+    return r
+
+
+def load_tokens(filename):
+    with open(filename, "r") as f:
+        return json.load(f)
+
+
+def filter_swaps(swaps, tokens):
+    return [
+        s for s in swaps
+        if {s['path'][0], s['path'][-1]} <= tokens
```

Comment:
What does <= tokens mean? shouldn't it be "in tokens"

Reply:
a in B means a ∈ B
A <= B means A ⊆ B

Reply:
Marco :heart: python sets!